core-bundle/src/Twig/Extension/ContaoExtension.php

```diff
@@ -172,7 +174,10 @@ public function renderLegacyTemplate(string $name, array $blocks, array $context
     {
         $template = Path::getFilenameWithoutExtension($name);
 
-        $partialTemplate = new class($template) extends FrontendTemplate {
+        $partialTemplate = new class($template) extends Template {
+            use FrontendTemplateTrait;
+            use BackendTemplateTrait;
```

Comment:
I don't like that we are mixing back end and front end methods here. Can we improve this somehow (e.g. by checking the template prefix for `be_`)?

Reply:
I don't think you can reliably find out the scope by the name. This might work for the core, but not in general.

Reply:
That was just an example. We could also add a `renderLegacyBackendTemplate()` and a `renderLegacyFrontendTemplate()` method to prevent mixing up the contexts.

Reply:
Not sure I understand. The template files themselves don't say anything about their scope - how would the proxy code know which one to call?

Reply:
AFAIK we would need to know if the template is a backend or frontend template at compile time to be able to call the correct function in https://github.com/contao/contao/blob/7bd09fae31188fb3d3a38c36e527e0ba5f6dfbf3/core-bundle/src/Twig/Interop/PhpTemplateProxyNode.php#L32-L40

I think extending from both traits is the best solution here.

Reply:
> AFAIK we would need to know if the template is a backend or frontend template at compile time

Even dynamically wouldn't work because the context doesn't tell you anything about the template class it was used with.